src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/CancelReindexRequestHandler.cs

```diff
@@ -63,7 +64,8 @@ public async Task<CancelReindexResponse> Handle(CancelReindexRequest request, Ca
                 // If the job is already completed for any reason, return conflict status.
                 if (outcome.JobRecord.Status.IsFinished())
                 {
-                    return new CancelReindexResponse(HttpStatusCode.Conflict, null);
+                    throw new RequestNotValidException(
+                        string.Format(Resources.ReindexJobInCompletedState, outcome.JobRecord.Id, outcome.JobRecord.Status));
```

Comment:
Might be good to add tests to cover this change in behavior?

Reply:
Added unit tests